### DISCUSSION TITLE
Atom compiling code forever

### DISCUSSION THREAD

**Discussion by sharath1562** [atom/atom#22677] [Q&A] [answered]
I just got started with java and  I am using atom code editor
when I run the script it just runs and runs but never gives output
you can see the below screenshot it is just a simple program but still
![Screenshot 2021-06-30 093833](https://user-images.githubusercontent.com/69711417/123900955-aeb76900-d987-11eb-981f-0736b1258586.png)

**Comment by suryaanshah** [comment, marked as answer]
Hey, thanks for reaching out!
My reproduction of your problem gave the following errors:
```
Main.java:2: error: class TakingUserInput is public, should be declared in a file named TakingUserInput.java
public class TakingUserInput {
       ^
```
1. The code gives an error for the reason that `TakingUserInput` is a public class but is not in any other `.java` file. That could be a contributing cause.
2. The core cause of this problem *could* be that atom is not configured with you computer yet.

I say *could* because this happened with me when I first started using atom. In my case, it was `python`. Please provide more info or play around with the settings.

Hope that helped. Thanks for reaching out!

**Reply by sharath1562** [reply]
This error only shows when I use the scanner function to take input

**Comment by ThatXliner** [comment]
Just saying, the script package does not support running programs that read from standard input. Maybe you're trying to read from standard input? (I do not know Java, but I'm guessing `new Scanner(System.in)` reads from standard input)

**Reply by suryaanshah** [reply]
In Java, the most popular way to read numbers from standard input is to use the Scanner class. @sharath1562 does this error show up with BufferedReader class?

**Reply by sharath1562** [reply]
Thank for helping me out.
I reinstalled the atom and downloaded a terminal package, now it works fine😊

**Reply by suryaanshah** [reply]
> Thank for helping me out.
> I reinstalled the atom and downloaded a terminal package, now it works fine😊

Always!
